common/ObjectUtils.js

```diff
@@ -0,0 +1,41 @@
+// Contains helpful methods for interacting with the Object data type.
+
+/**
+ * @notice Given `overrideProps` and `defaultProps` Objects, returns a new Object, `newObject`,
+ * with the same properties as `defaultProps`, but replaces any property values that overlap with
+ * those also contained in `overrideProps`. Performs validation tests on all `newObject` properties, as specified
+ * by `defaultProps`.
+ * @dev Throws an Error if any validation tests fail.
+ * @param {Object} [overrideProps] specifies property values in newly created Object that should differ from
+ * those contained in `defaultProps`.
+ * @param {Object{Object, Function}} defaultProps its properties will be the same as the newly created Object, but each
+ * property itself is an Object that has `value` and `isValid` properties. The `value` determine the default values of
+ * the new Object and `isValid` will be called to validate each of `newObject`'s properties.
+ * @return `newObject` a new Object with the same properties as `defaultProps`, or `defaultProps` if undefined `overrideProps`.
+ */
+const createObjectFromDefaultProps = (overrideProps, defaultProps) => {
+  if (!defaultProps) {
+    throw new Error("Undefined `defaultProps`");
+  }
+  if (!overrideProps) {
+    return defaultProps;
+  }
+
+  const newObject = {};
+
+  Object.keys(defaultProps).forEach(prop => {
+    // Set property value to that contained in `overrideProps` if it exists, else set to `defaultProps`.
+    newObject[prop] = prop in overrideProps ? overrideProps[prop] : defaultProps[prop].value;
+
+    // Validate property value, regardless if coming from `overrideProps` or `defaultProps`.
+    if (!defaultProps[prop].isValid(newObject[prop])) {
```

Comment:
Just a question so I understand: would this return false if the `isValid` prop is missing from the `defaultProps` object? Say for argument's sake that the default props in the `disputer` was missing this parm. would it fail?

Reply:
It would fail with someting like "cannot call isValid on object, or isValid is not a property of object". I can check for this more explicitly however

Reply:
great